doc/api/n-api.md

```diff
@@ -3269,9 +3269,11 @@ napi_status napi_create_async_work(napi_env env,
 - `[in] async_resource_name`: Identifier for the kind of resource that is
 being provided for diagnostic information exposed by the `async_hooks` API.
 - `[in] execute`: The native function which should be called to excute
-the logic asynchronously.
+the logic asynchronously. The given function is called from a worker pool
+thread and executes in parallel with the main event loop thread.
```

Comment:
@ebickle is it guaranteed to execute in parallel with the main event loop thread? Or is it more precise to say "can execute in parallel with the main event loop thread"? That is, is there a situation where the only thread available to do work in the thread pool is the main event loop thread?

Reply:
@digitalinfinity agree completely with the suggestion. Regardless of whether or not the `execute` function can ever run in the main event loop thread, it's never guaranteed to be executed in parallel due to the OS' context switching. Updated the documentation.